### DISCUSSION TITLE
Save button on top of nested forms

### DISCUSSION THREAD

**Discussion by lukuser** [Budibase/budibase#10059] [Help]
Hello, So I have a form with nested forms reflecting a data hierarchy of deal (many-one) property (many-one) address. The Property_Provider links to the Deal_provider via filter: (Property_Provider.id = binding Deal_repeater.Deal.FK_Property_id, and similar way between Property and Address.

Is there a way to have a button on top that saves all three forms?

Here's the form. The red arrow is for extra points - is there a way to move Deal.PartyRepresentedType to below Property/Space Type ?
![image](https://user-images.githubusercontent.com/81827963/225665410-e020ad1e-3886-479b-9892-b5df5b0617ec.png)

overall component structure:

top:
![image](https://user-images.githubusercontent.com/81827963/225667068-ea39b0f4-e355-4036-8aaa-7ac523d7cafd.png)
bottom:
![image](https://user-images.githubusercontent.com/81827963/225667608-e9ffec47-55b6-4b58-96dc-9c12111c2b5b.png)

Thanks,

**Comment by ConorWebb96** [comment]
Hey @lukuser,

Will just get the quick one at the end out of the way. To change the position of this all you would need to do is drag the Party represented field below the property type field.

Nesting forms always makes things a bit tricker. Does this need to be a nested form? If they are relationship fields within the current table you are saving, this likely won't need to be as it should link across whenever the save drafts button is clicked. 

Just a small note, when using the Budibase tables you won't be able to save multiple forms unless the button has access to the form, i.e the button is nested within all the forms. (From here you could chain the saving row actions on all the forms.) However, If you need to save multiple forms on a single button click and aren't using the Budibase tables you could achieve this through writing a custom query and then executing that whenever the button is clicked. (Image provided below.)

<img width="1600" alt="Screenshot 2023-03-20 at 08 37 23" src="https://user-images.githubusercontent.com/126772285/226287606-97bf7df4-39b3-411b-8663-625d54eeadab.png">

Could you provide a screenshot of your data for this table? Correct me if I'm wrong, but are the **Deal type**, **Party represented** and **property type** all relationships on the initial table? If this is the case you could save the whole form with the correct filtered data and it should work.

I hope this helps!
